app/models/current_order.rb

```diff
@@ -58,10 +61,10 @@ def paypal_url(current_user, return_url)
 
   def save_order(current_user)
     user = current_user
-    new_order = user.orders.new(:total => @total, :vat => vat, :delivery_cost => @delivery_cost, :invoice => @invoice, :Status => @status, :transaction_id => @transaction_id)
+    new_order = user.orders.new(:total => @total, :vat => vat, :delivery_cost => @delivery_cost, :invoice => @invoice, :Status => @status, :transaction_id => @transaction_id, :pickup_time => @pickup_time)  
     save_successful = new_order.save
     if save_successful
-      @ordered_items.each do |index, details|
+      @ordered_items["ordered_items"].each do |index, details|
```

Comment:
Unused block argument - `index`. If it's necessary, use `_` or `_index` as an argument name to indicate that it won't be used.